hs_core/hydroshare/resource.py

```diff
@@ -705,9 +705,10 @@ def delete_resource_file(pk, filename_or_id, user):
     else:
         raise ObjectDoesNotExist(filename_or_id)
 
-    if resource.raccess.public:
+    if resource.raccess.public or resource.raccess.discoverable:
         if not resource.can_be_public:
             resource.raccess.public = False
+            resource.raccess.discoverable = False
```

Comment:
You mean change the name of the property from 'can_be_public' to something like 'can_be_public_or_discoverable'